src/Raven.Server/Web/System/Processors/OngoingTasks/OngoingTasksHandlerProcessorForGetOngoingTasks.cs

```diff
@@ -297,11 +260,62 @@ protected override IEnumerable<OngoingTaskReplication> CollectExternalReplicatio
         }
     }
 
-    private OngoingTaskReplication GetExternalReplicationInfo(DatabaseTopology databaseTopology, ClusterTopology clusterTopology,
+    protected override IEnumerable<OngoingTaskPullReplicationAsHub> CollectPullReplicationAsHubTasks(TransactionOperationContext context, ClusterTopology clusterTopology, DatabaseRecord databaseRecord)
+    {
+        var pullReplicationHandlers = _database.ReplicationLoader.OutgoingHandlers.Where(n => n is OutgoingPullReplicationHandler).ToList();
+        foreach (var handler in pullReplicationHandlers)
+        {
+            var ex = handler.Destination as ExternalReplication;
+            if (ex == null) // should not happened
+                continue;
+
+            yield return GetPullReplicationAsHubTaskInfo(clusterTopology, ex);
+        }
+    }
+
+    protected override IEnumerable<OngoingTaskSubscription> CollectSubscriptionTasks(TransactionOperationContext context, ClusterTopology clusterTopology, DatabaseRecord databaseRecord)
+    {
+        foreach (var keyValue in ClusterStateMachine.ReadValuesStartingWith(context, SubscriptionState.SubscriptionPrefix(databaseRecord.DatabaseName)))
+        {
+            var subscriptionState = JsonDeserializationClient.SubscriptionState(keyValue.Value);
+            var tag = _database.WhoseTaskIsIt(databaseRecord.Topology, subscriptionState, subscriptionState);
+            OngoingTaskConnectionStatus connectionStatus;
+            if (tag != _server.NodeTag)
+            {
+                connectionStatus = OngoingTaskConnectionStatus.NotOnThisNode;
+            }
+            else if (_database.SubscriptionStorage.TryGetRunningSubscriptionConnectionsState(subscriptionState.SubscriptionId, out var connectionsState))
+            {
+                connectionStatus = connectionsState.IsSubscriptionActive() ? OngoingTaskConnectionStatus.Active : OngoingTaskConnectionStatus.NotActive;
+            }
+            else
+            {
+                connectionStatus = OngoingTaskConnectionStatus.NotActive;
+            }
+
+            yield return new OngoingTaskSubscription
+            {
+                // Supply only needed fields for List View
+                ResponsibleNode = new NodeId
+                {
+                    NodeTag = tag,
+                    NodeUrl = clusterTopology.GetUrlFromTag(tag)
+                },
+                TaskName = subscriptionState.SubscriptionName,
+                TaskState = subscriptionState.Disabled ? OngoingTaskState.Disabled : OngoingTaskState.Enabled,
+                MentorNode = subscriptionState.MentorNode,
+                TaskId = subscriptionState.SubscriptionId,
+                Query = subscriptionState.Query,
+                TaskConnectionStatus = connectionStatus
+            };
+        }
+    }
+
+    protected override ValueTask<OngoingTaskReplication> GetExternalReplicationInfoAsync(DatabaseTopology databaseTopology, ClusterTopology clusterTopology,
```

Comment:
We need to split those methods into 2 part.
part 1: get definition -> same code for sharded/non-sharded
part 2: get status -> this is a "local" info which we will implement only for the non-sharded version

Reply:
done